float64[] q
---
float64[] J